src/run.rs

```diff
@@ -227,6 +232,60 @@ pub(crate) fn run_repl(
         use_color,
     );
 
+    let delta = {
+        let name = "std".to_string();
+        let content = get_standard_library().as_bytes();
+
+        let mut working_set = StateWorkingSet::new(engine_state);
+
+        let start = working_set.next_span_start();
+        working_set.add_file(name.clone(), content);
+        let end = working_set.next_span_start();
+
+        let (_, module, _, _) = parse_module_block(
```

Comment:
Oh, and the panic you're seeing is probably caused by missing `working_set.add_block()`. You'll need to call that and pass the block returned from here to it. You can see an example in `use` implementation code: https://github.com/nushell/nushell/blob/a49e5b30ff5846b2cc88eaa9c210b1162872138e/crates/nu-parser/src/parse_keywords.rs#L2269

Reply:
i've tried something in fccbcb082, see my comment at the top where i explain how it behaves now :+1: